### PR TITLE
feat: multicall chunk size

### PR DESCRIPTION
**Pull request by jxom**
## PR-Codex overview
This PR adds a new `batchSize` parameter to `multicall` which limits the size of each calldata chunk. It also includes documentation updates and tests.

### Detailed summary
- Added `batchSize` parameter to `multicall` function
- Updated documentation for `multicall` and `publicClient`
- Added tests for `multicall` function with `batchSize` parameter and `publicClient` with `batchSize` parameter set in batch configuration

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/docs/contract/multicall.md [hunk 1]
```diff
@@ -223,6 +223,27 @@ const results = await publicClient.multicall({
 })
 ```
 
+### batchSize (optional)
+
+- **Type:** `number`
+- **Default:** [`client.batch.multicall.batchSize`](/docs/clients/public.html#batch-multicall-batchsize-optional) (if set) or `1024`
+
+The maximum size (in bytes) for each calldata chunk. Set to `0` to disable the size limit.
```
Comment (tmm) [inline site/docs/contract/multicall.md:231]: Worth popping in a sentence about what smaller versus larger numbers mean (like the trade off)?

File: src/actions/public/multicall.test.ts [hunk 2]
```diff
@@ -621,3 +622,47 @@ test('multicall contract deployed on later block', async () => {
     Version: viem@1.0.2"
   `)
 })
+
+test.only('stress', async () => {
```
Comment (tmm) [inline src/actions/public/multicall.test.ts:626]: Should we remove the `.only`?

**Review by tmm — COMMENTED** [review]
Changeset + coverage boost and lgtm

File: src/actions/public/multicall.test.ts (hunk 2 above)
Comment (jxom) [reply src/actions/public/multicall.test.ts:626]: oops

**Pull request merged**